comp/forwarder/defaultforwarder/forwarder.go

```diff
@@ -31,10 +33,36 @@ type provides struct {
 }
 
 func newForwarder(dep dependencies) provides {
-	options := dep.Params.CreationOptions(dep.Config, dep.Log)
+	options := creationOptions(dep.Params, dep.Config, dep.Log)
 	return NewForwarder(dep.Config, dep.Log, dep.Lc, true, options, dep.Params.UseNoopForwarder)
 }
 
+func creationOptions(params Params, config config.Component, log log.Component) *Options {
+	var options *Options
+	if !params.withResolver {
+		options = NewOptions(config, log, getMultipleEndpoints(config, log))
+	} else {
+		keysPerDomain := getMultipleEndpoints(config, log)
+		options = NewOptionsWithResolvers(config, log, resolver.NewSingleDomainResolvers(keysPerDomain))
+	}
```

Comment:
Can we switch boolean logic, I think it would make the code easier to read ?
i.e:
```
    if params.withResolver {
        keysPerDomain := getMultipleEndpoints(config, log)
		options = NewOptionsWithResolvers(config, log, resolver.NewSingleDomainResolvers(keysPerDomain))
	} else {
		options = NewOptions(config, log, getMultipleEndpoints(config, log))
	}
```

Reply:
I prefer to keep my version as the default case is `NewOptions` and not `NewOptionsWithResolvers`